dotnet/src/InternalUtilities/src/System/TypeConverterFactory.cs

```diff
@@ -51,6 +53,28 @@ internal static class TypeConverterFactory
             return converter;
         }
 
-        return null;
+        return new JsonSerializationTypeConverter(type);
+    }
+
+    internal sealed class JsonSerializationTypeConverter : TypeConverter
+    {
+        private Type type { get; }
+
+        public JsonSerializationTypeConverter(Type type)
+        {
+            this.type = type;
+        }
+
+        public override bool CanConvertFrom(ITypeDescriptorContext? context, Type sourceType) => true;
+
+        public override object? ConvertFrom(ITypeDescriptorContext? context, CultureInfo? culture, object value)
+        {
+            return JsonSerializer.Deserialize((string)value, this.type);
+        }
+
+        public override object? ConvertTo(ITypeDescriptorContext? context, CultureInfo? culture, object? value, Type destinationType)
+        {
+            return JsonSerializer.Serialize(value, this.type);
+        }
```

Comment:
I think you said you're going to sync with @SergeyMenshykh; I expect this is going to need to be changed based on his efforts around arbitrary object marshaling. As written, this expects everything is strings.

Reply:
had a sync with him and after some discussion we determined that this was fine. specifically because we should only ever be going to/from a string when marshalling a complex type.